### PR TITLE
[Merged by Bors] - [Async] Implement LogStream with async/await and combinators.

### PR DESCRIPTION
**Pull request by nlordell**
This PR implements log stream with async/await and stream combinators.

### TEst Plan

CI and unit tests.

### CODE REVIEW

File: generate/src/contract/events.rs [hunk 1]
```diff
@@ -300,14 +300,6 @@ fn expand_builder_type(event: &Event) -> Result<TokenStream> {
                 self
             }
 
-            /// Limit the number of events that can be retrieved by this filter.
-            ///
-            /// Note that this parameter is non-standard.
-            pub fn limit(mut self, value: usize) -> Self {
```
Comment (fleupold) [inline generate/src/contract/events.rs]: Why did the limit go?

**Review by fleupold — APPROVED** [review]
Looks good to me.

File: generate/src/contract/events.rs (hunk 1 above)
Comment (nlordell) [reply generate/src/contract/events.rs]: I think that was removed in a separate PR - you must have reviewed between rebasing.

**Review by e00E — APPROVED** [review]

File: generate/src/contract/events.rs (hunk 1 above)
Comment (nlordell) [reply generate/src/contract/events.rs]: I created #315 to add it back in. It got removed as part of a refactor, and it needed slightly different semantics now that we have our own filter type.

**Pull request closed without merging**